linear_iterator.go

```diff
@@ -94,6 +94,7 @@ func (it *linearIterator) Next() (*Connection, *Message, error) {
 				continue
 			case CompressionBZ2:
 				it.reader.Reset(bzip2.NewReader(bytes.NewReader(chunkData)))
+				it.inChunk = true
```

Comment:
All three cases set the same `true` value (except the default case). Could this be set always?

Reply:
indeed - will follow up with a cleanup and test PR shortly.